index.js

```diff
@@ -1,67 +1,200 @@
+        imageUrl: 'https://flxt.tmsimg.com/assets/p54287_p_v10_ad.jpg'
     },
     {
         title: 'Disclosure',
-        director: ['Sam feder']
+        director: 'Sam Feder',
+        year: 2020,
+        genre: ['Documentary'],
+        description: `An in-depth look at Hollywood's depiction of transgender people and the impact of those stories on transgender lives and American culture.`,
+        imageUrl: 'https://flxt.tmsimg.com/assets/p18415918_p_v13_aa.jpg'
     },
     {
         title: 'Passing',
-        director: ['Rebecca Hall']
+        director: 'Rebecca Hall',
+        year: 2021,
+        genre: ['Drama'],
+        description: `In 1920s New York City, a Black woman finds her world upended when her life becomes intertwined with a former childhood friend who's passing as white.`,
+        imageUrl: 'https://flxt.tmsimg.com/assets/p20588150_p_v10_aa.jpg'
     }
 ]
 
 app.use(morgan('common'));
 
-
-
 app.use(express.static('public',options));
 
 // GET requests
 app.get('/',(req,res)=>{
     res.send("<h1>This is just a default textual response. Nothing to worry about.</h1>")
 })
 
+// get all movies
 app.get('/movies',(req,res)=>{
-    res.json(tenMovies)
+    res.json(movies)
+})
+
+// get specific movie
+app.get('/movies/:title',(req,res)=>{
+    res.json(movies.find((movie)=>{
+        return movie.title ===req.params.title
+    }))
+})
+
+// get genre description by name
+app.get('/genres/:name',(req,res)=>{
+    res.send('This is a description about the film genre ' + req.params.name)
+})
+
+// get director data by name
+app.get('/directors/:name',(req,res)=>{
+    res.send('This is the data about ' + req.params.name)
+})
+
+// get all users
+app.get('/users',(req,res)=> {
+    res.json(users)
+})
+
+// get user by id
+app.get('/users/:id',(req,res)=>{
+    res.json(users.find((user)=>{
+        return req.params.id === user.id}));
+    })
+
+// add user 
+app.post('/users',(req,res)=>{
+    let newUser = req.body;
+
+    if (!newUser.username){
+        const message = 'Username is missing in request body';
+        res.status(400).send(message);
+    } else {  
+        newUser.id = uuid.v4();
+        users.push(newUser);
+        res.status(201).send('User added');
+}
+})
+
+// update username 
+app.put('/users/:id/:username',(req,res)=>{
+    let user = users.find((user)=>{
+        return user.id === req.params.id
+    });
+    if (user) {
+        user.username = req.params.username;
+        res.status(201).send('Username changed to ' + user.username)
+} else {
+    res.status(404).send('User not found.')
+}})
+
+// deregister user
+app.delete('/users/:username',(req,res)=>{
+    res.send('Your user was deleted!')
+})
```

Comment:
May we please implement this delete endpoint?

Reply:
done